active_learning_multiannotator/active_learning.ipynb

```diff
@@ -0,0 +1,677 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_accuacy_arr = np.full(num_rounds, np.nan)\n",
+    "\n",
+    "for i in range(num_rounds):\n",
+    "    # get consensus labels\n",
+    "    if i == 0:\n",
+    "        consensus_labels = get_majority_vote_label(multiannotator_labels)\n",
+    "    else:\n",
+    "        # we can use the pred_probs from last round as the best model estimate\n",
+    "        results = get_label_quality_multiannotator(multiannotator_labels, pred_probs_labeled, temp_scale=True)\n",
+    "        consensus_labels = results[\"label_quality\"][\"consensus_label\"]\n",
+    "\n",
+    "    # train model to get out-of-sample predicted probabilites \n",
+    "    pred_probs, pred_probs_unlabeled = fit_predict_proba(\n",
+    "        ExtraTreesClassifier(),\n",
+    "        X_labeled,\n",
+    "        consensus_labels,\n",
+    "        cv_n_folds=5,\n",
+    "        X_unlabeled=X_unlabeled,\n",
+    "    )\n",
+    "\n",
+    "    # train a model on the full set of labeled data to get the model accuracy for the current round \n",
+    "    # (this is an optional step used for demonstration purposed in this notebook, in practical settings it is unlikely to have ground truth labels)\n",
+    "    model = ExtraTreesClassifier()\n",
+    "    model.fit(X_labeled, consensus_labels)\n",
+    "    pred_labels = model.predict(X_test)\n",
+    "    model_accuacy_arr[i] = np.mean(pred_labels == true_labels_test)\n",
+    "\n",
+    "    # compute active learning scores\n",
+    "    active_learning_score, active_learning_score_unlabeled = get_active_learning_scores(\n",
+    "        multiannotator_labels, pred_probs, pred_probs_unlabeled\n",
+    "    )\n",
+    "\n",
+    "    # get the indices of examples to collect more labels for\n",
+    "    relabel_idx, relabel_idx_unlabeled = get_idx_to_label(\n",
+    "        active_learning_score=active_learning_score,\n",
+    "        active_learning_score_unlabeled=active_learning_score_unlabeled,\n",
+    "        batch_size_to_label=batch_size_to_label,\n",
+    "    )\n",
+    "\n",
+    "    # format the data for the next round of active learning \n",
+    "    # (moving some unlabeled examples to the labeled pool because we are collecting labels for them)\n",
+    "    (\n",
+    "        multiannotator_labels, relabel_idx_combined, X_labeled, X_unlabeled, pred_probs_labeled, \n",
+    "        pred_probs_unlabeled, extra_labels_labeled, extra_labels_unlabeled,\n",
+    "    ) = setup_next_iter_data(\n",
+    "        multiannotator_labels, relabel_idx, relabel_idx_unlabeled, X_labeled, X_unlabeled, pred_probs, \n",
+    "        pred_probs_unlabeled, extra_labels_labeled, extra_labels_unlabeled,\n",
+    "    )\n",
+    "\n",
+    "    # add a new annotator that provides new labels for the examples with lowest active learning scores (indices obtained above)\n",
```

Comment:
Note these long comments are getting cutoff for me in github's notebook viewer. So let's make sure all long comments are split into multiple lines and visible in the github viewer

Reply:
yep lgtm now, not sure what happened